lxd/storage/drivers/driver_btrfs.go

```diff
@@ -88,20 +89,21 @@ func (d *btrfs) load() error {
 // Info returns info about the driver and its environment.
 func (d *btrfs) Info() Info {
 	return Info{
-		Name:                  "btrfs",
-		Version:               btrfsVersion,
-		OptimizedImages:       true,
-		OptimizedBackups:      true,
-		OptimizedBackupHeader: true,
-		PreservesInodes:       !d.state.OS.RunningInUserNS,
-		Remote:                d.isRemote(),
-		VolumeTypes:           []VolumeType{VolumeTypeBucket, VolumeTypeCustom, VolumeTypeImage, VolumeTypeContainer, VolumeTypeVM},
-		BlockBacking:          false,
-		RunningCopyFreeze:     false,
-		DirectIO:              true,
-		IOUring:               true,
-		MountedRoot:           true,
-		Buckets:               true,
+		Name:                         "btrfs",
+		Version:                      btrfsVersion,
+		DefaultVMBlockFilesystemSize: deviceConfig.DefaultVMBlockFilesystemSize,
```

Comment:
Almost all references can be switched to the pools config. However the `lxd/project` package also has a reference to `DefaultVMBlockFilesystemSize` which causes an import loop when trying to resolve the driver within the `lxd/project` package (`lxd/storage` already imports from `lxd/project`).

Resolving the driver on the API handler level and passing it through doesn't sound to be a good idea.

But based on you comment I figured we need to have the logic in the `lxd/project` package anyway to pick the other value in case of the PowerFlex driver.

Reply:
@tomponline I tried several approaches in order to get `DefaultVMBlockFilesystemSize` unexported. The only propblem is the `lxd/project` package in which we would need to resolve the storage driver.
To not cause any import cycles my suggestion is to not return a struct from

```go
type Driver interface {
	// Internal.
	Info() Info <-- here
```

but instead use an interface. For example:

```go
type Driver interface {
	// Internal.
	Info() Informer
```

This means lot of refactoring in the rest of the code but would essentially allow to pass down the drivers return value of `Info()` into the project package to retrieve the value of `DefaultVMBlockFilesystemSize`. In `project/lxd` we would then make use of a package level interface that implements parts of `Informer` from the `lxd/storage` package.

Hope this makes sense :)

Reply:
OK lets leave it exported for now.

Reply:
Thanks, moved to another PR with https://github.com/canonical/lxd/pull/12566